atom/common/platform_util_win.cc

```diff
@@ -169,33 +170,298 @@ bool OpenExternal(const GURL& url) {
   return true;
 }
 
+
+// Required COM implementation of IFileOperationProgressSink so we can
+// precheck files before deletion to make sure they can be move to the 
+// Recycle Bin.
+class DelFileProgressSink : public IFileOperationProgressSink {
+  public:
+    DelFileProgressSink();
+    ~DelFileProgressSink(void);
+  private:
+	  ULONG STDMETHODCALLTYPE AddRef(void);
+    HRESULT STDMETHODCALLTYPE QueryInterface(REFIID   riid, LPVOID * ppvObj);
+    ULONG STDMETHODCALLTYPE Release( void);
+
+    HRESULT STDMETHODCALLTYPE StartOperations( void);
+    
+    HRESULT STDMETHODCALLTYPE FinishOperations( 
+    /* [in] */ HRESULT hrResult);
+    
+    HRESULT STDMETHODCALLTYPE PreRenameItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [string][unique][in] */ __RPC__in_opt_string LPCWSTR pszNewName);
+    
+    HRESULT STDMETHODCALLTYPE PostRenameItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
+    /* [string][in] */ __RPC__in_string LPCWSTR pszNewName,
+    /* [in] */ HRESULT hrRename,
+    /* [in] */ __RPC__in_opt IShellItem *psiNewlyCreated);
+    
+    HRESULT STDMETHODCALLTYPE PreMoveItem( 
+    /* [in] */ DWORD dwFlags,
+    /* [in] */ __RPC__in_opt IShellItem *psiItem,
```

Comment:
Definitely remove the __RPC__in_opt, unless they are running SAL (I doubt they are)